ni_measurementlink_service/_internal/discovery_client.py

```diff
@@ -132,15 +132,15 @@ def register_measurement_service(
                 )
             else:
                 _logger.exception("Error in registering with discovery service.")
-            return False
-        except FileNotFoundError:
+            raise grpc.RpcError(type(e).__name__, e.args[0]) from e
+        except FileNotFoundError as e:
             _logger.error(
                 "Unable to register with discovery service. Possible reason: discovery service not running."
             )
-            return False
-        except Exception:
+            raise FileNotFoundError(e.strerror, e.filename) from e
+        except Exception as e:
             _logger.exception("Error in registering with discovery service.")
-            return False
+            raise Exception(type(e).__name__, e.args[0]) from e
```

Comment:
To re-raise the current exception, write `raise` with no arguments.

Creating a new exception instance like this throws away information (derived type and backtrace), makes the error messages harder to understand, and is generally more complicated than it needs to be.

Reply:
Got it! I have only replace return False with raise. The sample exception is like this
```
2023-08-21 23:34:26,302 ERROR: Unable to register with discovery service. Possible reason: discovery service not running.
Traceback (most recent call last):
  File "Python\Python38\lib\runpy.py", line 194, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "Python\Python38\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File ".vscode\extensions\ms-python.python-2023.4.1\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher/../..\debugpy\__main__.py", line 39, in <module>
    cli.main()
  File ".vscode\extensions\ms-python.python-2023.4.1\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher/../..\debugpy/..\debugpy\server\cli.py", line 430, in main
    run()
  File ".vscode\extensions\ms-python.python-2023.4.1\pythonFiles\lib\python\debugpy\adapter/../..\debugpy\launcher/../..\debugpy/..\debugpy\server\cli.py", line 284, in run_file
    runpy.run_path(target, run_name="__main__")
  File ".vscode\extensions\ms-python.python-2023.4.1\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 321, in run_path
    return _run_module_code(code, init_globals, run_name,
  File ".vscode\extensions\ms-python.python-2023.4.1\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 135, in _run_module_code
    _run_code(code, mod_globals, init_globals,
  File ".vscode\extensions\ms-python.python-2023.4.1\pythonFiles\lib\python\debugpy\_vendored\pydevd\_pydevd_bundle\pydevd_runpy.py", line 124, in _run_code
    exec(code, run_globals)
  File "measurementlink-python\examples\sample_measurement\measurement.py", line 113, in 
<module>
    main()
  File "measurementlink-python\examples\sample_measurement\.venv\lib\site-packages\click\core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "measurementlink-python\examples\sample_measurement\.venv\lib\site-packages\click\core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "measurementlink-python\examples\sample_measurement\.venv\lib\site-packages\click\core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "measurementlink-python\examples\sample_measurement\.venv\lib\site-packages\click\core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "measurementlink-python\examples\sample_measurement\measurement.py", line 108, in 
main
    with sample_measurement_service.host_service():
  File "measurementlink-python\ni_measurementlink_service\measurement\service.py", line 379, in host_service
    self.grpc_service.start(
  File "measurementlink-python\ni_measurementlink_service\_internal\service_manager.py", 
line 105, in start
    self.discovery_client.register_measurement_service(port, service_info, measurement_info)
  File "measurementlink-python\ni_measurementlink_service\_internal\discovery_client.py", line 125, in register_measurement_service
    register_response = self.stub.RegisterService(request)
  File "measurementlink-python\ni_measurementlink_service\_internal\discovery_client.py", line 84, in stub
    address = _get_discovery_service_address()
  File "measurementlink-python\ni_measurementlink_service\_internal\discovery_client.py", line 220, in _get_discovery_service_address
    _ensure_discovery_service_started(key_file_path)
  File "measurementlink-python\ni_measurementlink_service\_internal\discovery_client.py", line 232, in _ensure_discovery_service_started
    exe_file_path = _get_discovery_service_location()
  File "measurementlink-python\ni_measurementlink_service\_internal\discovery_client.py", line 240, in _get_discovery_service_location
    registration_json_obj = json.loads(registration_json_path.read_text())
  File "Python\Python38\lib\pathlib.py", line 1236, in read_text
    with self.open(mode='r', encoding=encoding, errors=errors) as f:
  File "Python\Python38\lib\pathlib.py", line 1222, in open
    return io.open(self, mode, buffering, encoding, errors, newline,
  File "Python\Python38\lib\pathlib.py", line 1078, in _opener
    return self._accessor.open(self, flags, mode)
FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Program Files\\National Instruments\\Shared\\MeasurementLink\\MeasurementLinkService.json'
```